src/OpenTelemetry.Exporter.Jaeger/JaegerExporterHelperExtensions.cs

```diff
@@ -40,17 +41,55 @@ public static TracerProviderBuilder AddJaegerExporter(this TracerProviderBuilder
             {
                 return deferredTracerProviderBuilder.Configure((sp, builder) =>
                 {
-                    AddJaegerExporter(builder, sp.GetOptions<JaegerExporterOptions>(), configure);
+                    AddJaegerExporter(builder, sp.GetOptions<JaegerExporterOptions>(), configure, sp);
                 });
             }
 
-            return AddJaegerExporter(builder, new JaegerExporterOptions(), configure);
+            return AddJaegerExporter(builder, new JaegerExporterOptions(), configure, serviceProvider: null);
         }
 
-        private static TracerProviderBuilder AddJaegerExporter(TracerProviderBuilder builder, JaegerExporterOptions options, Action<JaegerExporterOptions> configure = null)
+        private static TracerProviderBuilder AddJaegerExporter(
+            TracerProviderBuilder builder,
+            JaegerExporterOptions options,
+            Action<JaegerExporterOptions> configure,
+            IServiceProvider serviceProvider)
         {
             configure?.Invoke(options);
 
+            if (options.Protocol == JaegerExportProtocol.HttpBinaryThrift && options.HttpClientFactory == null)
+            {
+                if (serviceProvider != null)
+                {
+                    options.HttpClientFactory = () =>
+                    {
+                        Type httpClientFactoryType = Type.GetType("System.Net.Http.IHttpClientFactory, Microsoft.Extensions.Http", throwOnError: false);
```

Comment:
What's the typical number of invocations expected for this factory? If more than once, perhaps extract the reflection logic out.

Reply:
Just once during construction of the exporter instance: https://github.com/open-telemetry/opentelemetry-dotnet/blob/5fd75ba23eb43049dae0d31dd109088fbef7e18f/src/OpenTelemetry.Exporter.Jaeger/JaegerExporter.cs#L70